shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -274,6 +277,20 @@ - (void)replaceRange:(UITextRange*)range withText:(NSString*)text {
 }
 
 - (BOOL)shouldChangeTextInRange:(UITextRange*)range replacementText:(NSString*)text {
+  NSRange currentRange = ((FlutterTextRange*)range).range;
+  if ( (currentRange.location > 0 && [text length] > 0 &&
```

Comment:
Off the top of my head, I can't remember if we can ever be passed a range with location `NSNotFound` but many other methods in `UITextInput` are required to guard against bad inputs. I don't see anything in Apple's docs for this method, but it seems prudent to guard against that.